src/tools/tidy/src/extdeps.rs

```diff
@@ -14,6 +15,19 @@ const ALLOWED_SOURCES: &[&str] = &[
 /// workspace `Cargo.toml`.
 pub fn check(root: &Path, bad: &mut bool) {
     for &(workspace, _, _) in crate::deps::WORKSPACES {
+        let gitmodules = root.join(".gitmodules");
+        let submodules = build_helper::util::parse_gitmodules(&gitmodules);
```

Comment:
Can we move this out of the loop? No need to re-parse for every workspace.

Same with the other call site in deps.